src/metadata/ColumnMetadata.ts

```diff
@@ -656,10 +657,21 @@ export class ColumnMetadata {
      * Compares given entity's column value with a given value.
      */
     compareEntityValue(entity: any, valueToCompareWith: any) {
-        const columnValue = this.getEntityValue(entity);
+        let columnValue = this.getEntityValue(entity);
```

Comment:
Not sure this is the proper way to handle this, however as long as we use it only in RelationIdLoader for comparing primary keys and return to the user fully transformed entities we should be ok.
In the future if we decide to use this function for something else we might have some problems. This lines will also be changed once we separate library core from driver implementations, but it's problem for another day.

Reply:
yeah Im not sure about these changes too. Probably I'll need to check myself this issue.

Reply:
I wasn't thrilled about this change either. However, this was the most appropriate location I could find for the functionality, however I'm far from expert on typeorm's inner workings. Very happy to defer the fix for this one to either of you two.